items/propm_allocation.py

```diff
@@ -0,0 +1,190 @@
+#!python3
+
+"""
+An implementation of a PROPm allocation algorithm. Reference:
+
+    Artem Baklanov, Pranav Garimidi, Vasilis Gkatzelis, and Daniel Schoepflin (2021).
+    ["PROPm Allocations of Indivisible Goods to Multiple Agents"](https://arxiv.org/abs/2105.11348).
+
+Programmer: Maksim Timokhin
+Since:  2021-05
+"""
+
+import networkx as nx
+import numpy as np
+from fairpy import valuations, ValuationMatrix, Allocation
+from typing import List, Set
+from copy import deepcopy
+
+
+def insert_agent_into_allocation(agent: int, item: int, allocated_bundles: List[List[int]]):
+    """
+    If agent's i value of item j is greater than 1/n, we can allocate item j to i and solve
```

Comment:
Can you please add some doctests to each function? This will greatly help understanding them. Thanks!

Reply:
Done